zerver/tests/test_unread.py

```diff
@@ -137,3 +137,21 @@ def test_update_all_flags(self):
         for msg in self.get_old_messages():
             self.assertEqual(msg['flags'], [])
 
+    def test_mark_all_in_stream_read(self):
+        self.login("hamlet@zulip.com")
+        test_msg = self.send_message("hamlet@zulip.com", "Denmark", Recipient.STREAM, "test")
+
+
+        result = self.client.post("/json/messages/flags", {"messages": ujson.dumps([]),
+                                                           "op": "add",
+                                                           "flag": "read",
+                                                           "stream_name": "Denmark"})
+        self.assert_json_success(result)
+        user_messages = list(UserMessage.objects.filter(message=test_msg))
+        self.assertEqual(len(user_messages) > 0, True)
+        for um in user_messages:
+            if um.user_profile.email != "hamlet@zulip.com":
+                self.assertFalse(um.flags.read)
+            else:
+                self.assertTrue(um.flags.read)
```

Comment:
Maybe worth also adding an assert that messages sent to another stream by that user weren't marked as read.